hasher-matcher-actioner/hmalib/lambdas/aws_secrets.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import boto3
+import base64
+import typing as t
+import os
+import ast
+from botocore.exceptions import ClientError
+
+from threatexchange.api import ThreatExchangeAPI
+
+AWS_REGION = os.environ["AWS_REGION"]
+
+class AWSSecrets():
+    """
+    A class for reading secrets stored in aws for interacting with ThreatExchange
+    """
+    def te_api_data(self) -> "ThreatExchangeAPI" :
+        secret_name = "threatexchange/api_keys"
+        secret_str = get_str_secret(secret_name)
+        return ThreatExchangeAPI(secret_str)
```

Comment:
Alt, ignorable: a common pattern is a that alternative constructors are stored as classmethods

```
ThreatExchangeAPI:
  def __init__(self, x):
    self.x = x

  @classmethod
  def fromSecret(cls):
    return cls(get_str_secret("threatexchange/api_keys"))

api = ThreatExchangeAPI::fromSavedSecret
```

Reply:
Would second this. At present, the separation of concerns between secrets and the API is unclear. I would go further and have the inference being done at callsite.

eg.
```python
te_secret = hmalib.secrets.get_secret('threatexchange/secret')
api = ThreatExchangeAPI(te_secret)
```

That way, ThreatexchangeAPI does not need to know we are fetching secrets from AWS and hmalib.aws_secrets need not know we are using secrets to create ThreatExchangeAPI objects.